nmdc_runtime/api/db/mongo.py

```diff
@@ -101,6 +103,20 @@ def activity_collection_names(mdb: MongoDatabase) -> Set[str]:
     }
 
 
+@lru_cache
+def planned_process_collection_names(mdb: MongoDatabase) -> Set[str]:
+    schema_view = nmdc_schema_view()
+    collection_names = set()
+    class_name_to_collection_names = defaultdict(set)
+    for collection_name, class_names in collection_name_to_class_names.items():
+        for class_name in class_names:
+            class_name_to_collection_names[class_name].add(collection_name)
+    for descendant in schema_view.class_descendants("PlannedProcess"):
+        collection_names |= class_name_to_collection_names[descendant]
+
+    return nmdc_schema_collection_names(mdb) & collection_names
```

Comment:
My initial thought here was that the operand on the left-hand side was redundant. Looking at its definition above (not changed in this diff), I see it takes into account whether a collection is empty or not. I don't know how/why that characteristic is relevant to this function.